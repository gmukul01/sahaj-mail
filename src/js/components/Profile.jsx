import React from 'react';
import PropTypes from 'prop-types';

import Avatar from 'components/Avatar';

const Profile = ({ userName }) => (
	<div className="profile">
		<Avatar />
		<p>{userName}</p>
	</div>
);

Profile.propTypes = {
	userName: PropTypes.string
};

export default React.memo(Profile);
