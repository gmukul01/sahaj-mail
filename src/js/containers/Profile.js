import { connect } from 'react-redux';

import Profile from 'components/Profile';

const mapStateToProps = ({ user: { name } }) => ({
	userName: name
});

export default connect(
	mapStateToProps,
	null
)(Profile);
