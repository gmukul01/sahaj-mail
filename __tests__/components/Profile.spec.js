import React from 'react';
import renderer from 'react-test-renderer';

import Profile from 'components/Profile';

describe('Profile Component', () => {
	it('should match snapshot', () => {
		const profile = renderer.create(<Profile userName="DummyName" />).toJSON();
		expect(profile).toMatchSnapshot();
	});
});
