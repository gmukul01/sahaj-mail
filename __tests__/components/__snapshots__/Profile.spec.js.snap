// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component should match snapshot 1`] = `
<div
  className="profile"
>
  <div
    className="avatar"
  >
    <img
      alt="User Pic"
      src="http://i.pravatar.cc/150?img=66"
    />
  </div>
  <p>
    DummyName
  </p>
</div>
`;
